core/domain/question_jobs_one_off.py

```diff
@@ -0,0 +1,99 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for questions."""
+
+import logging
+
+from constants import constants
+from core import jobs
+from core.domain import question_domain
+from core.domain import question_services
+from core.platform import models
+import feconf
+
+(question_models,) = models.Registry.import_models([models.NAMES.question])
+
+
+class QuestionMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-time job that may be used to migrate question schema
+    versions. This job will load all existing questions from the data store
+    and immediately store them back into the data store. The loading process of
+    a question in question_services automatically performs schema updating.
+    This job persists that conversion work, keeping questions up-to-date and
+    improving the load time of new questions.
+    """
+
+    _DISABLED_KEY = 'new_structures_disabled'
+    _DELETED_KEY = 'question_deleted'
+    _ERROR_KEY = 'validation_error'
+    _MIGRATED_KEY = 'question_migrated'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [question_models.QuestionModel]
+
+    @staticmethod
+    def map(item):
+        if not constants.ENABLE_NEW_STRUCTURES:
+            yield (
+                QuestionMigrationOneOffJob._DISABLED_KEY,
```

Comment:
Handle this in the same way as "deleted", since this is an expected thing.

Only yield IDs for unexpected error cases (so that we can debug).

(Ditto elsewhere.)

Reply:
I thought about this, but it doesn't seem very useful since when these structures are released, these guards should simply be removed(and before that, they would not trigger since there exist no datastores). Doing this would just increase the amount of code we need to remove on a later date. Is that intended or am I misunderstanding something?

Reply:
That's a very fair point. I was going according to the general principle of "aggregate if expected in practice", but I realize from your comment that this is not expected in practice! So on balance I think that what you have now is fine, and I like your reasoning. Thank you for clarifying!